### PR TITLE
    resmgr: handle UpdateContainer requests better.

### PR DESCRIPTION
**Pull request by klihub**
Note: this PR is stacked on top of #252.

Try doing a better job of detecting and handling redundant UpdateContainer requests which do not really change any of a container's parameters. Such requests are typically sent by an active CPU Manager with the static policy enabled in kubelet.
    
Before evaluating an update request for redundancy fill in any missing values from the container's current resources. Then if the request looks redundant short-circuit handling it by sending a reply with the current resources. Otherwise pass it on to the policy.

### CODE REVIEW

File: pkg/resmgr/nri.go [hunk 1]
```diff
@@ -432,13 +432,33 @@ func (p *nriPlugin) UpdateContainer(ctx context.Context, pod *api.PodSandbox, co
...
+		}
+		if v := c.GetCpusetCpus(); v != "" {
+			c.SetCpusetCpus(v)
+		}
+		if v := c.GetCpusetMems(); v != "" {
+			c.SetCpusetMems(v)
+		}
+		if v := c.GetMemoryLimit(); v != 0 {
+			c.SetMemoryLimit(v)
+		}
```
Comment (askervin) [inline pkg/resmgr/nri.go:453]: Would it be good to short-circuit MemorySwap, too? Just for symmetry, if nothing else.

**Review by askervin — COMMENTED** [review]
One question, otherwise LGTM.

**Review by askervin — APPROVED** [review]
LGTM

**Review by fmuyassarov — APPROVED** [review]
lgtm

**Pull request merged**